lab.R

```diff
@@ -133,63 +149,121 @@ problem3.4 = 937
-#' 
+# They are respresented by a sinlge number that appears to be like a scale from 1-10.  
 #' 
 #' 
 
 #' 2. *Since ultimately we're going to construct a Spearman rank correlation matrix (the quick-and-dirty approach to the problem), we need to get `exter.cond` into an integer representation. First, let's generate a table that shows the distribution across values of `exter.cond`, using `dplyr::count()`. Call this `ex_cond_count`. We'll use this to check the conversion process over the next few steps.*
-# ex_cond_count = ???
+ex_cond_count = dataf_nodup%>%
+   dplyr::count(exter.cond)
 
 #' 3. *As a first attempt, write a function `char_to_int()` that takes a character vector as input, coerces to a factor using `as.factor()`, and then coerces it to `as.integer()`.  Using `mutate()` and `count()`, apply this function to `exter.cond` and check the distribution against your answer for #2.* 
+#char_to_int() into character vector, then to a factor with as.factor, then eventually into an integer with as.integer
+#how to write a function from happycoding:
+ #1 Write the return type of the function.
+ #2 Write the name of the function.
+ #3 Inside parenthesis (), list any parameters the function takes.
+ #4 Inside curly brackets {}, write the code that will run whenever the function is called. This is called the body of the function.
+ char_to_int= function(character_vector) { #name and return type 
+   charactor_factor = as.factor(character_vector) #taking vector and turning into a factor
+   factor_integer= as.integer(charactor_factor) #taking factor and turning it into integer
+ }
+ #trying to apply it to external condition, using mutate and count
+dataf_nodup %>%
+  mutate(exter.cond= char_to_int(exter.cond)) %>%
+  count(exter.cond)
+   
+
+
 
 #' 4. *Can you explain what went wrong?  Hint:  Check the docs for the `levels` argument of `factor()`.*
+#' So it "works" in terms of it does write a function (which I was pleasantly surprised I figured out how to do after many attempts!) but it doesn't put the data in the right format, or in the format that actually ranks them by quality. I think it is just showing them in 1-5 order based on their alphabetical standing.
 #' 
 #' 
 #' 
 
 #' 5. *We can fix this by passing in a character vector of the levels in the desired order, namely, from Po (Poor) to Ex (Excellent).  Modify `char_to_int()` to use such a vector in the `as.factor()` call.  Why doesn't this work?*
 #' 
-#' 
-#' 
+ # establish the order
+character_order <- c('Po', 'Fa', 'Ta', 'Gd', 'Ex') # establish the order
+char_to_int = function(character_vector, character_order= c('Po', 'Fa', 'Ta', 'Gd', 'Ex')) { #name and return type 
+  charactor_factor = as.factor(character_vector, levels=character_order) #taking vector and turning into a factor, and trying to add a section for adding levels to this
```

Comment:
The error message is 
```
 Error: Problem with `mutate()` column `exter.cond`.
ℹ `exter.cond = char_to_int(exter.cond, character_order)`.
✖ unused argument (levels = character_order)
```
This is confusing because the problem is happening lower in the call stack.  The clue is `levels = character_order`.  You don't have an argument like this on line 200.  Instead it's here, line 190.  `as.factor()` doesn't take an argument named `levels`, and this is what's causing the error. 
```suggestion
  charactor_factor = as.factor(character_vector) #taking vector and turning into a factor, and trying to add a section for adding levels to this
```
This will run without an error, but it won't put the levels in the order we want.  That's fixed in the next step.